### ISSUE TITLE
Error messages could be better in case of an unreachable database

### ISSUE THREAD

**Issue by Alwaysin** [FreshRSS/FreshRSS#4813]
I had a problem which lead to my database being down.

Symptoms were quite far off the expected ones:
- red flag "A request has failed, it may have been caused by internet connection problems." on top of the opened webpage
- HTTP ERROR 500 when reloading the webpage
- at the same time, no log in /data/users/_/*
- at the same time, logs in Apache2:
```
[Fri Nov 04 14:27:08.282930 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:11919] AH01071: Got error 'PHP message: PHP Warning:  Undefined array key 1 in /var/www/html/freshrss/lib/lib_rss.php on line 821'
[Fri Nov 04 14:27:08.283032 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:11919] malformed header from script 'index.php': Bad header: \t</body></html>
[Fri Nov 04 14:27:08.283106 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:11919] AH01070: Error parsing script headers
[Fri Nov 04 14:27:08.283122 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] (22)Invalid argument: [client x.x.x.x:11919] AH01075: Error dispatching request to :
[Fri Nov 04 14:28:01.653278 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:12224] AH01071: Got error 'PHP message: PHP Warning:  Undefined array key 1 in /var/www/html/freshrss/lib/lib_rss.php on line 821'
[Fri Nov 04 14:28:01.653382 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:12224] malformed header from script 'index.php': Bad header: \t</body></html>
[Fri Nov 04 14:28:01.653478 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] [client x.x.x.x:12224] AH01070: Error parsing script headers
[Fri Nov 04 14:28:01.653636 2022] [proxy_fcgi:error] [pid 44370:tid 139728266151488] (22)Invalid argument: [client x.x.x.x:12224] AH01075: Error dispatching request to :
```

None of these helped me to understand that the database was down.

**To Reproduce**
Steps to reproduce the behavior: just stop mysql/mariadb/postgresql db engine.

**Expected behavior**
Message indicating that the database is not reachable.

**Comment by math-GH** [comment]
Is it still reproduce-able?

**Comment by Alwaysin** [comment]
Yes.

**Comment by Alkarex** [comment]
Some of the reported errors were fixed in https://github.com/FreshRSS/FreshRSS/pull/5039 and released in FreshRSS 1.21.0.
Could you please give a try to our newest FreshRSS 1.21.0 (or edge)?

I just did a test with a MySQL database and get this:

![image](https://user-images.githubusercontent.com/1008324/222956676-0eea3eda-a814-4a34-8157-bbe04e114e7f.png)

and

![image](https://user-images.githubusercontent.com/1008324/222955617-7bdef928-9a1d-45cc-8d00-3795050cc635.png)

**Comment by Alwaysin** [comment]
I always run on edge, and I don't have this behaviour. Probably related to something on my side then, so let's close it.
